Tasks/NuGetCommand/nugetcommandmain.ts

```diff
@@ -6,20 +6,29 @@ import * as nugetPublish from './nugetpublisher';
 import * as nugetPack from './nugetpack';
 import * as nugetCustom from './nugetcustom';
 import nuGetGetter = require("nuget-task-common/NuGetToolGetter");
+import * as commandHelper from "nuget-task-common/CommandHelper";
+import semver = require('semver');
 
 const NUGET_EXE_CUSTOM_LOCATION: string = "NuGetExeCustomLocation";
 
 async function main(): Promise<void> {
     tl.setResourcePath(path.join(__dirname, "task.json"));
 
+    logNugetStartupVariables();
+
     // Getting NuGet
     tl.debug('Getting NuGet');
     let nuGetPath: string = undefined;
     try {
         nuGetPath = tl.getVariable(nuGetGetter.NUGET_EXE_TOOL_PATH_ENV_VAR) || tl.getVariable(NUGET_EXE_CUSTOM_LOCATION);
-        if (!nuGetPath){
-            nuGetGetter.cacheBundledNuGet_4_0_0();
-            nuGetPath = await nuGetGetter.getNuGet("4.0.0");
+        if (!nuGetPath) {
+            let cachedVersionToUse = nuGetGetter.DEFAULT_NUGET_VERSION;
+            nuGetGetter.cacheBundledNuGet();
```

Comment:
Not obligatory: But it'd be cool if cacheBundledNuGet returned the path to the cached location of NuGet so we can avoid checking the env var here too.